tests/system.rs

```diff
@@ -85,3 +85,25 @@ fn mut_system() {
         world.run_systems();
     }
 }
+
+#[test]
+fn scheduler_resource() {
+    let world = World::default();
+
+    let mut state = 5_u32;
+
+    let immut_resource = "foo".to_string();
+
+    let scheduler = Scheduler::default();
+
+    scheduler.register(|_world, state| {
+        *state += 1;
+        assert_eq!(immut_resource, "foo");

```

Comment:
here's an example of a system referring to something outside the system without capturing it